### PR TITLE
add `archive.closeCheckout` method

### PR DESCRIPTION
**Pull request by poga**
fix https://github.com/mafintosh/hyperdrive/issues/241

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -108,11 +109,17 @@ class Hyperdrive extends EventEmitter {
...
-    self.metadata.on('extension', extension)
-    self.metadata.on('peer-add', peeradd)
-    self.metadata.on('peer-remove', peerremove)
+    let listeners = {
+      'error': onerror,
+      'append': update,
+      'extension': extension,
+      'peer-add': peeradd,
+      'peer-remove': peerremove
+    }
```
Comment (mafintosh) [inline index.js]: @andrewosh I'm wondering fs there any need in a checkout to actually listen for these events?

Comment (poga) [reply index.js]: well if they're not needed for a checkout, that's even better! no need for the weird `closeCheckout()` method. :D

Comment (andrewosh) [reply index.js]: Yeah I'd agree that checkouts shouldn't be adding listeners for these events. We should prob do a check to see if the metadata matches what's passed in as an option (which is what a checkout does) before attaching listeners.

**Comment by andrewosh** [comment]
Closing because this has been fixed in later releases.

**Pull request closed without merging**